File: tembu-server/tembu-server/Models/LicenseInfo.cs
namespace tembu_server.Models;

public class LicenseInfo
{
    public bool IsValid { get; set; }
    public string Email { get; set; } = "";
    public DateTime ExpiryDate { get; set; }
    public string ErrorMessage { get; set; } = "";

    public bool IsExpired => DateTime.Today > ExpiryDate;
    public int DaysRemaining => Math.Max(0, (ExpiryDate - DateTime.Today).Days);
}

File: tembu-server/Models/LicenseInfo.vb
Public Class LicenseInfo

    Public Property IsValid As Boolean
    Public Property Email As String
    Public Property ExpiryDate As DateTime
    Public Property MaxUsers As Integer
    Public Property ErrorMessage As String

    Public ReadOnly Property IsExpired As Boolean
        Get
            Return DateTime.Today > ExpiryDate
        End Get
    End Property

    Public ReadOnly Property DaysRemaining As Integer
        Get
            Return Math.Max(0, (ExpiryDate - DateTime.Today).Days)
        End Get
    End Property

End Class
